### PR TITLE
Fix: CPU hot busyloop by setting a default timeout to 30s, if not set

### PR DESCRIPTION
**Pull request by szuecs**
Signed-off-by: Sandor Szücs <sandor.szuecs@zalando.de>

### CODE REVIEW

**Comment by szuecs** [comment]
:+1:

File: net/httpclient.go [hunk 1]
```diff
@@ -80,6 +84,10 @@ func NewTransport(options Options) *Transport {
 		options.Tracer = &opentracing.NoopTracer{}
 	}
 
+	if options.Timeout == 0 {
+		options.Timeout = defaultTimeout
```
Comment (aryszka) [inline net/httpclient.go]: let's only enforce the IdleConnTimeout. The other settings have sane defaults.

Comment (szuecs) [reply net/httpclient.go]: ok

Comment (szuecs) [reply net/httpclient.go]: fixed

**Comment by mikkeloscar** [comment]
:+1:

**Comment by szuecs** [comment]
:+1:

**Pull request merged**